pkg/services/container/morph/executor.go

```diff
@@ -335,22 +336,45 @@ func (s *morphExecutor) validateToken(t *sessionV2.Token, cIDV2 *refs.ContainerI
 		return fmt.Errorf("incorrect token signature: %w", err)
 	}
 
-	if cIDV2 == nil { // can be nil for PUT or wildcard may be true
+	if cIDV2 == nil { // can be nil for PUT
 		return nil
 	}
 
-	if sessionCID := cc.ContainerID().GetValue(); !bytes.Equal(sessionCID, cIDV2.GetValue()) {
-		return fmt.Errorf("wrong container: %s", base58.Encode(sessionCID))
-	}
+	var cIDRequested cid.ID
 
-	var cID cid.ID
-
-	err = cID.ReadFromV2(*cIDV2)
+	err = cIDRequested.ReadFromV2(*cIDV2)
 	if err != nil {
 		return fmt.Errorf("invalid container ID: %w", err)
 	}
 
-	cnr, err := s.rdr.Get(cID)
+	if cc.Wildcard() {
+		keyFromToken := crypto.UnmarshalPublicKey(t.GetSignature().GetKey())
+		if keyFromToken == nil {
+			return errors.New("error while decoding public key from the token's signer")
+		}
+
+		userFromToken := user.ResolveFromECDSAPublicKey(*keyFromToken)
+
+		cIDs, err := s.rdr.List(&userFromToken)
```

Comment:
`s.rdr.Get(cIDRequested)` below is good for this

Reply:
initially there was another context so yes, fixed